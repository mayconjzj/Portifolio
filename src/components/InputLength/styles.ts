import tw from 'tailwind-styled-components';

export const Content = tw.div`
  flex
  flex-col
  w-[200px]
`;
