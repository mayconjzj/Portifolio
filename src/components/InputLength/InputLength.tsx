'use client';

import { useInputLength } from './hooks';
import * as S from './styles';

export const InputLength = () => {
  const { caractersLength, setCharactersLength } = useInputLength('15');

  return (
    <S.Content>
      <span>
        Tamanho: <span>{caractersLength}</span> Caracteres
      </span>

      <input
        type="range"
        min={5}
        max={25}
        value={caractersLength}
        onChange={(e) => setCharactersLength(e.target.value)}
      />

      <button>Gerar Senha</button>
    </S.Content>
  );
};
