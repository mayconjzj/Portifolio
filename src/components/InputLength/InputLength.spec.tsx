import { render } from '@testing-library/react';

import { InputLength } from './InputLength';

describe('InputLength', () => {
  it('should render successfully', () => {
    const { baseElement, asFragment } = render(<InputLength />);
    expect(baseElement).toBeTruthy();

    expect(asFragment()).toMatchSnapshot();
  });
});
