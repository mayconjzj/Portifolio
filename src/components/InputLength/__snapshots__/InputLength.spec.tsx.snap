// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLength should render successfully 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col w-[200px]"
  >
    <span>
      Tamanho: 
      <span>
        15
      </span>
       Caracteres
    </span>
    <input
      max="25"
      min="5"
      type="range"
      value="15"
    />
    <button>
      Gerar Senha
    </button>
  </div>
</DocumentFragment>
`;
